chirps_utility.py

```diff
@@ -0,0 +1,140 @@
+from chirps import *
+from main import *
+from csv_utility import *
+from user_utility import *
+from random_id_generator import *
+
+class ChirpsUtility:
+
+	def view_chirps(public_convos, private_convos, chirps_dict, current_user):
+		"""
+		Prints out all public and private chirps that the current user is a part of
+
+		Arg- chirps dictionary (from CSV file), currently selected user
+		"""
+		# counter and empty list to be used for convo select
+		# (display counter as choice number, append each convo to convo_id_list,
+		# convo int(counter - 1) to find convo in convo_id_list by index
+		counter = 1
+		convo_id_list = list()
+		# if a user is signed in
+		if current_user != None:
+			try:
+				print("\n ~ PRIVATE CHIRPS ~ ")
+				for convo_key, chirps_list in private_convos.items():
+					# chirp info that started convo
+					convo_start = chirps_list[0]
+					# if current user is the recipient of a private chirp
+					if chirps_dict[convo_start][1] == current_user.screen_name:
+						print("{0}. {1}: {2}".format(counter, chirps_dict[convo_start][0], chirps_dict[convo_start][2]))
+						convo_id_list.append(convo_key, "private")
+						counter += 1
+					# if current user sent a private chirp
+					elif chirps_dict[convo_start][0] == current_user.screen_name:
+						print("{0}. to {1}: {2}".format(counter, chirps_dict[convo_start][1], chirps_dict[convo_start][2]))
+						convo_id_list.append([convo_key, "private"])
+						counter += 1
+				# print all public chirps
+				return ChirpsUtility.print_public_chirps(public_convos, convo_id_list, chirps_dict, counter)
+			except AttributeError:
+				return ChirpsUtility.print_public_chirps(public_convos, convo_id_list, chirps_dict, counter)
+			except KeyError:
+				return ChirpsUtility.print_public_chirps(public_convos, convo_id_list, chirps_dict, counter)
+		else:
+			return ChirpsUtility.print_public_chirps(public_convos, convo_id_list, chirps_dict, counter)
```

Comment:
Since your try, all your excepts, _and_ your else return the same method call, you can drop the `else` entirely, unindent the else's `return` to the same level the else was at, remove the `return` entirely from the `try`, and stick a `pass` as the body of both the `except` blocks.  That way, all paths end up at that single return without duplicating code 4x